diff --git a/svgpathtools/path.py b/svgpathtools/path.py
index 83250a0..434aef5 100644
--- a/svgpathtools/path.py
+++ b/svgpathtools/path.py
@@ -327,7 +327,7 @@ def transform(curve, tf):
         invT = np.linalg.inv(tf[:2,:2])
         D = reduce(np.matmul, [invT.T, Q, invT])
 
-        eigvals, eigvecs = np.linalg.eig(D)
+        eigvals, eigvecs = np.linalg.eigh(0.5*(D+D.T))  # symmetrized in case of floating point error; D already symmetric
 
         rx = 1 / np.sqrt(eigvals[0])
         ry = 1 / np.sqrt(eigvals[1])
