diff --git a/src/cmake/thirdparty/SetupHDF5.cmake b/src/cmake/thirdparty/SetupHDF5.cmake
index afe59bc..3aee1ee 100644
--- a/src/cmake/thirdparty/SetupHDF5.cmake
+++ b/src/cmake/thirdparty/SetupHDF5.cmake
@@ -75,6 +75,7 @@ else()
                  HINTS ${HDF5_DIR}/cmake/hdf5 
                        ${HDF5_DIR}/lib/cmake/hdf5
                        ${HDF5_DIR}/share/cmake/hdf5
+                       ${HDF5_DIR}/share/hdf5
                        ${HDF5_DIR}/cmake)
 endif()
 
