--- fmt/include/fmt/format.h	2025-07-01 17:48:22.672881000 -0700
+++ format.h	2025-07-01 17:48:23.250317000 -0700
@@ -1624,9 +1624,14 @@
   return {multiply(x.f, y.f), x.e + y.e + 64};
 }
 
+// BEGIN AXOM BUGFIX
 template <typename T, bool doublish = num_bits<T>() == num_bits<double>()>
 using convert_float_result =
-    conditional_t<std::is_same<T, float>::value || doublish, double, T>;
+    conditional_t<std::is_same<T, float>::value ||
+                      std::numeric_limits<T>::digits ==
+                          std::numeric_limits<double>::digits,
+                  double, T>;
+// END AXOM BUGFIX
 
 template <typename T>
 constexpr auto convert_float(T value) -> convert_float_result<T> {
