diff --git a/include/fmt/format.h b/include/fmt/format.h
index 3f8507f7..be75b891 100644
--- a/include/fmt/format.h
+++ b/include/fmt/format.h
@@ -1624,9 +1624,14 @@ FMT_CONSTEXPR inline auto operator*(fp x, fp y) -> fp {
   return {multiply(x.f, y.f), x.e + y.e + 64};
 }
 
+// BEGIN AXOM BUGFIX
 template <typename T, bool doublish = num_bits<T>() == num_bits<double>()>
 using convert_float_result =
-    conditional_t<std::is_same<T, float>::value || doublish, double, T>;
+    conditional_t<std::is_same<T, float>::value ||
+                      std::numeric_limits<T>::digits ==
+                          std::numeric_limits<double>::digits,
+                  double, T>;
+// END AXOM BUGFIX
 
 template <typename T>
 constexpr auto convert_float(T value) -> convert_float_result<T> {
