diff --git a/include/RAJA/internal/ThreadUtils_CPU.hpp b/include/RAJA/internal/ThreadUtils_CPU.hpp
index fe262363a..76aa7ba0b 100644
--- a/include/RAJA/internal/ThreadUtils_CPU.hpp
+++ b/include/RAJA/internal/ThreadUtils_CPU.hpp
@@ -22,7 +22,10 @@
 #include "RAJA/config.hpp"
 
 #include "RAJA/pattern/thread.hpp"
+#if defined(RAJA_ENABLE_OPENMP)
+#include <omp.h>
 #include "RAJA/policy/openmp/thread.hpp"
+#endif
 #include "RAJA/policy/sequential/thread.hpp"
 
 namespace RAJA
