diff --git a/include/RAJA/policy/openmp/thread.hpp b/include/RAJA/policy/openmp/thread.hpp
index faf58d38b..dcbd555b3 100644
--- a/include/RAJA/policy/openmp/thread.hpp
+++ b/include/RAJA/policy/openmp/thread.hpp
@@ -22,6 +22,11 @@
 
 #if defined(RAJA_OPENMP_ACTIVE)
 
+// Include OpenMP header for Windows
+#if defined(_OPENMP)
+#include <omp.h>
+#endif
+
 #include "RAJA/util/macros.hpp"
 
 #include "RAJA/policy/thread_auto.hpp"
