From 05a3b56b0c2df32c6ea02df6af28a1b292502408 Mon Sep 17 00:00:00 2001
From: Cyrus Harrison <cyrush@llnl.gov>
Date: Tue, 25 Jul 2017 06:18:44 -0700
Subject: [PATCH] compile grammer files during setup

---
 setup.py | 26 ++++++++++++++++++++++++++
 1 file changed, 26 insertions(+)

diff --git a/setup.py b/setup.py
index a35c769..b3af739 100644
--- a/setup.py
+++ b/setup.py
@@ -4,6 +4,7 @@ from setuptools import setup, find_packages
 import os
 import sys
 from distutils import log
+from distutils.cmd import Command
 
 import sphinx
 
@@ -171,6 +172,31 @@ else:
 
     cmdclass['compile_catalog'] = compile_catalog_plusjs
 
+class CompileGrammarCommand(Command):
+    description = 'Compile python grammar file for pycode'
+    user_options = []
+
+    def initialize_options(self):
+        pass
+
+    def finalize_options(self):
+        pass
+
+    def run(self):
+        from sphinx.pycode.pgen2.driver import compile_grammar
+
+        compile_grammar('sphinx/pycode/Grammar-py2.txt')
+        print('sphinx/pycode/Grammar-py2.txt ... done')
+
+        compile_grammar('sphinx/pycode/Grammar-py3.txt')
+        print('sphinx/pycode/Grammar-py3.txt ... done')
+
+    def sub_commands(self):
+        pass
+
+
+cmdclass['compile_grammar'] = CompileGrammarCommand
+
 
 setup(
     name='Sphinx',
-- 
2.9.2

