diff --git a/thirdparty_builtin/CMakeLists.txt b/thirdparty_builtin/CMakeLists.txt
index 83db75b..c0cbd92 100644
--- a/thirdparty_builtin/CMakeLists.txt
+++ b/thirdparty_builtin/CMakeLists.txt
@@ -112,9 +112,15 @@ if(ENABLE_TESTS)
         # For more info see: https://github.com/LLNL/blt/issues/79
         # 
         
-        blt_append_custom_compiler_flag( FLAGS_VAR gtest_extra_flags
-                                         DEFAULT " " 
-                                         MSVC  "/wd4251" "/wd4275" )
+        if(MSVC AND ENABLE_CUDA)
+          set(gtest_extra_flags
+              "$<$<NOT:$<COMPILE_LANGUAGE:CUDA>>:/wd4251>"
+              "$<$<NOT:$<COMPILE_LANGUAGE:CUDA>>:/wd4275>")
+        else()
+          blt_append_custom_compiler_flag( FLAGS_VAR gtest_extra_flags
+                                           DEFAULT " "
+                                           MSVC  "/wd4251" "/wd4275" )
+        endif()
 
         set (INSTALL_GMOCK OFF CACHE BOOL "")
         set (INSTALL_GTEST OFF CACHE BOOL "")
