diff -c cmake-3.7.2/Modules/FindPythonLibs.cmake cmake-3.7.2-patch/Modules/FindPythonLibs.cmake 
*** cmake-3.7.2/Modules/FindPythonLibs.cmake	2017-01-13 06:05:41.000000000 -0800
--- cmake-3.7.2-patch/Modules/FindPythonLibs.cmake	2017-01-31 14:28:21.000000000 -0800
***************
*** 168,174 ****
  
      # Use the library's install prefix as a hint
      set(_Python_INCLUDE_PATH_HINT)
!     get_filename_component(_Python_PREFIX ${PYTHON_LIBRARY} PATH)
      get_filename_component(_Python_PREFIX ${_Python_PREFIX} PATH)
      if(_Python_PREFIX)
        set(_Python_INCLUDE_PATH_HINT ${_Python_PREFIX}/include)
--- 168,181 ----
  
      # Use the library's install prefix as a hint
      set(_Python_INCLUDE_PATH_HINT)
!     # get_filename_component(_Python_PREFIX ${PYTHON_LIBRARY} PATH)
!     # the above fails when find_package(PythonInterp) returns a list with 
!     # both the python lib and a symlink to it.
!     # 
!     # prevent this case by only using the first item in ${PYTHON_LIBRARY}
!     # as input to get_filename_component
!     list(GET ${PYTHON_LIBRARY} 0 FIRST_FOUND_PYTHON_LIBRARY)
!     get_filename_component(_Python_PREFIX ${FIRST_FOUND_PYTHON_LIBRARY} PATH)
      get_filename_component(_Python_PREFIX ${_Python_PREFIX} PATH)
      if(_Python_PREFIX)
        set(_Python_INCLUDE_PATH_HINT ${_Python_PREFIX}/include)